python/cudf/dataframe/series.py

```diff
@@ -1483,27 +1488,21 @@ def label_encoding(self, cats, dtype=None, na_sentinel=-1):
         -------
         A sequence of encoded labels with value between 0 and n-1 classes(cats)
         """
+        from cudf import DataFrame
 
-        if self.null_count != 0:
-            mesg = 'series contains NULL values'
-            raise ValueError(mesg)
+        value = Series(cats, dtype=self.dtype)
+        order = Series(cudautils.arange(len(self)))
+        codes = Series(cudautils.arange(len(value)))
 
-        if self.dtype.kind not in 'iuf':
-            raise TypeError('expecting integer or float dtype')
+        if dtype is not None:
+            codes = codes.astype(dtype)
```

Comment:
You can actually pass a dtype to the `cudautils.arange(...)` call instead of typecasting afterwards.

Reply:
The More You Know ™